packages/search/src/search.tsx

```diff
@@ -84,366 +99,412 @@ type Props = {
+		hideOpenIcon,
+		openIconSide,
+		minLength,
+		maxLength,
+		hideClose,
+	}: Props,
+	forwardedRef: Ref< ImperativeHandle >
+): JSX.Element => {
+	const { __ } = useI18n();
+	const [ keyword, setKeyword ] = React.useState( defaultValue );
+	const [ isOpen, setIsOpen ] = React.useState( defaultIsOpen );
+	const [ hasFocus, setHasFocus ] = React.useState( autoFocus );
+
+	const instanceId = useInstanceId( InnerSearch, 'search' );
+	const searchInput: MutableRefObject< HTMLInputElement | null > = React.useRef( null );
+	const openIcon: RefObject< HTMLButtonElement | null > = React.useRef( null );
+	const overlay: RefObject< HTMLDivElement | null > = React.useRef( null );
+
+	const focus = React.useCallback( () => {
+		// if we call focus before the element has been entirely synced up with the DOM, we stand a decent chance of
+		// causing the browser to scroll somewhere odd. Instead, defer the focus until a future turn of the event loop.
+		setTimeout( () => searchInput.current?.focus(), 0 );
+	}, [] );
+
+	const blur = React.useCallback( () => searchInput.current?.blur(), [] );
+
+	const clear = React.useCallback( (): void => setKeyword( '' ), [] );
+
+	React.useImperativeHandle(
+		forwardedRef,
+		() => ( {
+			focus,
+			blur,
+			clear,
+		} ),
+		[ focus, blur, clear ]
+	);
+
+	const doSearch: ( ( search: string ) => void ) & { cancel?: () => void } = React.useMemo(
+		() => ( delaySearch ? debounce( onSearchProp, delayTimeout ) : onSearchProp ),
+		[ onSearchProp, delayTimeout, delaySearch ]
+	);
+
+	useEffect( () => {
+		if ( keyword ) {
+			onSearchProp( keyword );
 		}
+		// Disable reason: This effect covers the case where a keyword was passed in as the default value and we only want to run it on first search; the useUpdateEffect below will handle the rest of the time that keyword updates
+		// eslint-disable-next-line react-hooks/exhaustive-deps
+	}, [] );
 
-		// if there's a keyword change: trigger search
-		if ( this.state.keyword ) {
-			this.onSearch( this.state.keyword );
+	useUpdateEffect( () => {
+		if ( keyword ) {
+			doSearch( keyword );
 		} else {
-			// explicitly bypass debouncing when there is no keyword is empty
-			if ( this.props.delaySearch ) {
+			// explicitly bypass debouncing when keyword is empty
+			if ( delaySearch ) {
 				// Cancel any pending debounce
-				this.onSearch.cancel?.();
+				doSearch.cancel?.();
 			}
 
-			this.props.onSearch( this.state.keyword );
+			doSearch( keyword );
 		}
 
-		this.props.onSearchChange?.( this.state.keyword );
-	}
+		onSearchChange?.( keyword );
+	}, [ doSearch, keyword, onSearchChange ] );
+
+	const openSearch = React.useCallback(
+		(
+			event:
+				| MouseEvent< HTMLButtonElement | HTMLInputElement >
+				| KeyboardEvent< HTMLButtonElement | HTMLInputElement >
+		): void => {
+			event.preventDefault();
+
+			setKeyword( '' );
+			setIsOpen( true );
+
+			focus();
+			// no need to call `onSearchOpen` as it will be called by `onFocus` once the searcbox is focused
```

Comment:
If `onSearchOpen` is now called only on focus, it means it never gets the `event` parameter. It can be removed from the type.